### PR TITLE
Feature 1300/alter template backend update

### LINKED ISSUE #1300: BE: Alter template backend to not have edit endpoint (XS)
Depends on #1298 

## Definition of Done
- [x] Make template not editable
- [x] Make template questions not editable
- [x] Update tests accordingly

### PR DESCRIPTION
**Pull request by pieperm**
Closes #1300 

This feature is another installment of the major refactors started in issues #1296 and #1298. The goal of this issue is to remove the ability to update certain fields of templates, including the template title and description. 

The only field that can be updated is `active`, which is a method for soft-deleting templates. When a template is soft-deleted, it will still show up as part of past feedback requests, but it will be inaccessible for creating new feedback requests (unless it is reactivated)

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateServicesImpl.java [hunk 1]
```diff
@@ -106,32 +106,30 @@ public FeedbackTemplate getById(UUID id) {
...
     @Override
     public List<FeedbackTemplate> findByFields(@Nullable UUID creatorId, @Nullable String title) {
         if (!getIsPermitted()) {
             throw new PermissionException("You are not authorized to do this operation");
         }
 
         List<FeedbackTemplate> templateList = new ArrayList<>();
-        // Filters only active templates by default
+
         if (title != null) {
```
Comment (bhatiam12) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateServicesImpl.java:117]: This logic can be simplified if you write custom query in repository method. then you don't need to use retainall or addall at all and query will do all the work. Take a look at the query in OpportunitiesRepository.java. We can discuss if you have questions.

**Comment by bhatiam12** [comment]
This has same code coverage issue, we will need to spend some time to add tests otherwise it will keep happening. @mkimberlin Please can we take a look at quality gates and see how we can tweak UI test coverage.

File: server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateServicesImpl.java [hunk 2]
```diff
@@ -99,53 +82,22 @@ public FeedbackTemplate getById(UUID id) {
...
-        if (!getIsPermitted()) {
-            throw new PermissionException("You are not authorized to do this operation");
-        }
 
         return feedbackTemplate.get();
     }
 
     @Override
     public List<FeedbackTemplate> findByFields(@Nullable UUID creatorId, @Nullable String title) {
-        if (!getIsPermitted()) {
```
Comment (bhatiam12) [inline server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateServicesImpl.java:112]: why did we remove this validation ?

Comment (smithj11) [reply server/src/main/java/com/objectcomputing/checkins/services/feedback_template/FeedbackTemplateServicesImpl.java:112]: getIsPermitted only checks if there is an authenticated user. However, we determined that the backend throws an error if there is not an authenticated user before it even hits this service implementation, making it a moot point.

**Review by bhatiam12 — APPROVED** [review]

**Review by moshirim — APPROVED** [review]

**Pull request merged**